invenio_records_rest/__init__.py

```diff
@@ -22,7 +22,634 @@
+  take the record unique identifiers and return FetchedPID objects, which
+  consist of the Persistent Identifier and the record data.
+  The fetcher for each endpoint in Invenio-Records-REST can be configured
+  and it is defined by its name, default is ``recid``. To register a new
+  fetcher, add the ``invenio_pidstore.fetchers`` entrypoint in the
+  ``setup.py`` of a module and point it to the fetcher function. For more
+  information on fetchers, see the corresponding paragraph in
+  Invenio-PIDStore.
+
+  The fetchers by default will query the database, but this can be redirected
+  to Elasticsearch according to the needs of the application. More
+  specifically, the fetcher takes as a parameter a provider object,
+  which is responsible for the querying. In order to change the behavior of
+  the GET one would write a new provider class to
+  override the ``get()`` method of the ``BaseProvider`` and query
+  Elasticsearch.
+
+Serialization
+-------------
+A key feature of invenio-records-rest is the ability to transform
+records from JSON to other formats. Through this process the content
+of records can be enriched, removed of sensitive information when required
+or be export in standardized bibliographic formats like DataCite, DublinCore.
+
+Content negotiation
+~~~~~~~~~~~~~~~~~~~
+The serializer which will be used is chosen by the
+MIME type specified in the headers of the request. There can be different
+serializers for the multiple results returned by a search query, and the
+single result returned when querying for specific records.
+This can be configured in the following part of the records rest endpoints:
+
+>>> RECORDS_REST_ENDPOINTS = dict(
+...    recid=dict(
+...       record_serializers=dict(
+...          application/json=('invenio_records_rest.serializers'
+...                            ':json_v1_response'),
+...       ),
+...       search_serializers=dict(
+...          application/json=('invenio_records_rest.serializers'
+...                            ':json_v1_search'),
+...       )
+...    )
+... )
+
+With the usage of MIME types it is also possible to specify the
+version of a given serialization, as serialization schemas can have
+newer versions over time. More specifically, if we have created two
+serializers, one for DataCite's version 3.1 metadata schema and one
+for version 4.1, we can specify which one we require in the
+Content-Type http header field. For example, if we have
+set in the configuration:
+
+>>> record_serializer={
+...    'application/datacite.v3.1+xml': ('invenio_records_rest.serializers'
+...                                      'datacite_response_v31'),
+...    'application/datacite.v4.1+xml': ('invenio_records_rest.serializers'
+...                                      'datacite_response_v41'),
+... }
+
+
+Then we can send an http request with the http headers containing
+``Content-Type:application/datacite+xml`` or
```

Comment:
I don't know if plain `application/datacite+xml` would work... Maybe what is worth giving an example of is somethings like this:
```python
record_serializer={
    "application/datacite+xml": "invenio_records_rest.serializers.datacite_response_v31",
    "application/datacite.v3.1+xml": "invenio_records_rest.serializers.datacite_response_v31",
    "application/datacite.v4.1+xml": "invenio_records_rest.serializers.datacite_response_v41",
}
```
...where if asking for `application/datacite+xml` (or plain `application/datacite`) would return the v3.1 serialization.